### ISSUE TITLE
Provide documentation on how to use web workers for Dart 2

### ISSUE THREAD

**Issue by mnordine** [dart-lang/site-www#4749]
Dart SDK Version 2  
Windows, MacOS  
Chrome  

I decided to try out Dart 2, and one of the first things I wanted to look into is how to run workers on the web, since `Isolate` is no longer available for the web. Here is the documentation: https://api.dartlang.org/stable/2.0.0/dart-html/Worker-class.html  

There is no indication whatsoever on how to proceed on such an important web feature, given the ever-increasing importance of concurrent computation on modern systems.  

Some questions that immediately spring to mind:
- JS Worker and Dart Worker take a script url
- How does that work?
- Do I pass in a Dart url?
-  If so, what does that Dart file look like?
- Can I pass in a function instead of a script url?  

In a more general sense, Dart documentation for web APIs is not a good experience. Are those docs simply generated from browser IDL?  Those docs should contain API overview, working examples for each function where not immediately obvious, links to articles, etc...Almost every page I visit is just method and property names. There's no guidance for the programmer.

**Comment by antfitch** [comment]
Tentative closing this one since web workers don't appear to work with the Dart Development Compiler right now. When support is added, a topic would make sense. If I'm missing something, feel free to create a new issue since this one is quite old.

**Comment by mnordine** [comment]
When is support being added?

**Comment by antfitch** [comment]
I would ask that question in https://github.com/dart-lang/sdk/issues/35730

**Comment by sigmundch** [comment]
My quick 2 cents - I believe the DDC support is not as important to unblock this, and we can deliver useful documentation even without it.

To clarify a little bit:
* There are Dart APIs in the deprecated `dart:html` and in the newer `pacakge:web` related to web workers. Those APIs are automatically derived from IDLs and are JS-centric. As a result, the documentation is hard to link back to how it relates to Dart.
* It is possible to write Dart code that accesses those APIs (DDC and dart2js) and it is possible to run Dart programs as workers (dart2js only). Since teams deploy with dart2js, that is sufficient to get a system working.

In terms of what should the documentation focus on. I believe one of the biggest challenges for developers is how to put their Dart code into a web worker and interact with it from the page. I think it will help to frame it in 3 stages:
* (a) how to use a JavaScript web worker from Dart (focuses on APIs available from the main-page)
* (b) how to use a Dart web worker, compiled with dart2js, from a JavaScript page (focuses on how to compile Dart as a worker script, and how to use APIs available to workers)
* (c) finally, how to use a Dart web worker from a Dart program. I believe it is best to treat (c) as a combination of (a) and (b). Each program (page or worker) should treat the other as if it was written in JavaScript, even though it was compiled from Dart. That framing helps answer many of the other questions that developers come across (e.g. URLs to use refer to deployment JS urls for the compiled code, serialization between the two should be done assuming that the other program doesn't share any logic with ours, just like we do with JS-interop across programs on the same page).

cc @biggs0125 as FYI
